### PR TITLE
Add "SVG" support to download options 

### PR DESCRIPTION
**Pull request by dogukanuhn**
Hello,

I saw an issue named "[BUG]: export png is blurry and unreadable compared to web UI" and i thought, there should be and SVG option.

It is working fine in my local but maybe there should be some css edits. 

image

### CODE REVIEW

**Comment by dogukanuhn** [comment]
https://github.com/AykutSarac/jsoncrack.com/issues/194

File: src/components/FileInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  &.active {
+    background: ${({ theme }) => theme.PRIMARY};
+    color: white;
+    outline: 3px solid ${({ theme }) => theme.BACKGROUND_TERTIARY};
+    border-radius: 10px;
+  }
+`;
+
+export interface ModalProps {
+    value:string | number | readonly string[] | undefined,
```
Comment (AykutSarac) [inline src/components/FileInput/index.tsx:50]: Please change it to `value?: string | number | readonly string[]`

File: src/components/FileInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    extensions:string[]
+    activeExtension:number
+    setExtension: Function,
+    onChange: React.ChangeEventHandler<HTMLInputElement> | undefined,
```
Comment (AykutSarac) [inline src/components/FileInput/index.tsx:54]: here too

File: src/components/FileInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    outline: 3px solid ${({ theme }) => theme.BACKGROUND_TERTIARY};
+    border-radius: 10px;
+  }
+`;
+
+export interface ModalProps {
+    value:string | number | readonly string[] | undefined,
+    extensions:string[]
+    activeExtension:number
+    setExtension: Function,
```
Comment (AykutSarac) [inline src/components/FileInput/index.tsx:53]: Function is too general decleration, can we use `setExtension: (...params) => return type` here

File: src/containers/Modals/DownloadModal/index.tsx [hunk 4]
```diff
@@ -137,12 +143,14 @@ export const DownloadModal: React.FC<ModalProps> = ({ visible, setVisible }) =>
...
-      const dataURI = await toPng(imageElement, {
+      let exportImage = extension === Extensions.svg ? toSvg : toPng
+
+      const dataURI = await exportImage(imageElement, {
         quality: fileDetails.quality,
         backgroundColor: fileDetails.backgroundColor,
       });
 
-      downloadURI(dataURI, `${fileDetails.filename}.png`);
+      downloadURI(dataURI, `${fileDetails.filename}.${Extensions[extension]}`);
```
Comment (AykutSarac) [inline src/containers/Modals/DownloadModal/index.tsx:153]: Would `${fileDetails.filename}.${extension}` do the job here?

File: src/containers/Modals/DownloadModal/index.tsx [hunk 5]
```diff
@@ -162,10 +170,12 @@ export const DownloadModal: React.FC<ModalProps> = ({ visible, setVisible }) =>
         <StyledContainer>
           File Name
           <StyledColorWrapper>
-            <Input
-              placeholder="File Name"
+            <FileInput
               value={fileDetails.filename}
               onChange={e => updateDetails("filename", e.target.value)}
+              setExtension={(ext:number)=> setExtension(ext)}
```
Comment (AykutSarac) [inline src/containers/Modals/DownloadModal/index.tsx:176]: Let's use `setExtension={setExtension}` here

File: src/containers/Modals/DownloadModal/index.tsx [hunk 6]
```diff
@@ -162,10 +170,12 @@ export const DownloadModal: React.FC<ModalProps> = ({ visible, setVisible }) =>
...
+              activeExtension={extension}
+              extensions={Object.keys(Extensions).filter((v) => isNaN(Number(v)))}
```
Comment (AykutSarac) [inline src/containers/Modals/DownloadModal/index.tsx:178]: Could it be written in more simpler way?

**Review by AykutSarac — CHANGES_REQUESTED** [review]

**Pull request closed without merging**